fabtests/pytest/efa/test_unexpected_msg.py

```diff
@@ -1,7 +1,19 @@
 import pytest
+from efa.efa_common import efa_run_client_server_test
 
+@pytest.mark.parametrize("iteration_type",
+                         [pytest.param("short", marks=pytest.mark.short),
+                          pytest.param("standard", marks=pytest.mark.standard)])
 @pytest.mark.functional
-def test_unexpected_msg(cmdline_args):
-    from common import ClientServerTest
-    test = ClientServerTest(cmdline_args, "fi_unexpected_msg -e rdm -I 10")
-    test.run()
+def test_unexpected_msg(cmdline_args, iteration_type, completion_semantic, memory_type, message_size):
+    if completion_semantic == "delivery_complete":
+        pytest.skip("Unexpected message test does not support delivery_complete completion semantic")
+
+    # SHM provider currently supports 1 in flight message for host<->device transfers
+    if memory_type in ["cuda_to_host", "host_to_cuda"]:
+        inflight_msgs = 1
+    else:
+        inflight_msgs = 4
```

Comment:
won't this fail for platform without CMA? Or you'd like to skip the test in the caller of fabtests

Reply:
You're right. Can't think of a good way to skip this just for CMA disabled. I think it's best to wait until we fix SHM to merge this.